core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -82,6 +75,33 @@ def start_navigator(self) -> None:
     def start_serial(self, device: str) -> None:
         self.start_mavlink_manager(Endpoint("Master", self.settings.app_name, "serial", device, 115200, protected=True))
 
+    def run_with_sitl(self, vehicle: str = "vectored") -> None:
+        firmware = os.path.join(self.settings.firmware_path, "sitl")
+        if not os.path.exists(firmware):
+            temporary_file = self.firmware_download.download(Vehicle.Sub, Platform.SITL)
+            assert temporary_file, "Failed to download SITL binary."
+            shutil.move(str(temporary_file), firmware)
+
+        # ArduPilot SITL binary will bind TCP port 5760 (server) and the mavlink router will connect to it as a client
+        master_endpoint = Endpoint("Master", self.settings.app_name, "tcpin", "127.0.0.1", 5760, protected=True)
```

Comment:
can we use `Endpoint.TCPServer` ?
Is that going to work right on both mavproxy and mavlinkrouter? I expected TCPClient here

Reply:
Yes. I was just replicating how it is currently, to change them all together, but I will append a commit changing that on this PR.

Reply:
It may be necessary to use UDP here, if we send a reboot command the TCP client/server logic may fail since it connects only once and it does not reconnect alone.
@Williangalvani you are the network guy, can you confirm this ?
It may be safer to use UDP for our master.

Reply:
Ok, but is this related to sitl or are you suggesting to change it for all other masters?

Reply:
Yes, depending on the implementaton on the router side, it may just give up after the connection is lost and never reconnect.
We should move to UDP if possible or at least make sure that the routers reconnect on ardusub restart.
This may affect SITL and Navigator.

Reply:
Just to point it, I found this problem while working around #170

Reply:
Apparently there's no way to output SITL to UDP, according to [the ardupilot docs](https://ardupilot.org/dev/docs/using-sitl-for-ardupilot-testing.html#sitl-with-mavproxy-udp) and the `--help`:

![image](https://user-images.githubusercontent.com/6551040/120378612-0618d980-c2f5-11eb-94c0-a1b724fd8711.png)

Reply:
try uartA as udpin

Reply:
```Starting sketch 'ArduSub'
Starting SITL input
Using Irlock at port : 9005
PANIC: Invalid device path: udpin
Running: sh dumpstack.sh 89024 >dumpstack_sitl.89024.out 2>&1
Failed```

Reply:
while that should technically work, I'm afraid that some things could be output only to the main port.